### PR TITLE
Clock stop event

### PR DESCRIPTION
**Pull request by dwastberg**
Add an onStop Event to the Clock which is called when the clock reaches its stopTime.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @dwastberg! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://api.github.com/repos/AnalyticalGraphicsInc/cesium/contents/CONTRIBUTORS.md) file!
* :x: Missing CLA.
    * Please send in a [Contributor License Agreement](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) and comment back here to let us know to check this!
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/clock_stop_event/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by dwastberg** [comment]
CLA has been sent

**Comment by pjcozzi** [comment]
Thanks again for the contribution, @dwastberg.  We received your CLA.

**Comment by hpinkos** [comment]
Thanks @dwastberg!  This PR looks fine to me.  Can you please explain a scenario where this event is useful?

**Comment by dwastberg** [comment]
I'm using the clock to control a looping simulation and at the end of each loop I want to clean up some objects placed in the scene and make some changes to my GUI.

File: Specs/Core/ClockSpec.js [hunk 1]
```diff
@@ -394,6 +394,47 @@ defineSuite([
...
+            clockStep : ClockStep.TICK_DEPENDENT,
+            multiplier : multiplier,
+            clockRange : ClockRange.LOOP_STOP,
+            shouldAnimate : true
+        });
+        var onStopSpy = jasmine.createSpy('event');
+        clock.onStop.addEventListener(onStopSpy);
+        clock.tick();
+        expect(onStopSpy).toHaveBeenCalled();
+
```
Comment (hpinkos) [inline Specs/Core/ClockSpec.js]: remove this whitespace

**Comment by hpinkos** [comment]
Thanks @dwastberg!  Just that one comment.  And please add a note under the `Additions` section of `CHANGES.md`

**Comment by dwastberg** [comment]
Just wondering if there is anything else needed to merger this?

**Comment by hpinkos** [comment]
Thanks for the reminder @dwastberg!  We don't get notifications when you add more commits, so it's a good idea to make a comment to let us know when a PR is ready for another look.

This all looks good to me.  
Does anyone else have feedback?  If not, I'll merge this by the end of the day

**Comment by hpinkos** [comment]
Thanks again @dwastberg!

**Pull request merged**